app/src/main/java/com/group80/uoftinder/UserScoreCalculator.java

```diff
@@ -0,0 +1,54 @@
+package com.group80.uoftinder;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+
+public class UserScoreCalculator {
+    private ArrayList<HashSet<Integer>> userAnswers;
+    private final boolean[] isMultiSelect;
+    private final int[] answerBitLengths;
+    private final int answerLen;
+
+    public UserScoreCalculator(ArrayList<HashSet<Integer>> userAnswers, boolean[] isMultiSelect, int[] answerBitLengths) {
+        this.userAnswers = userAnswers;
+        this.isMultiSelect = isMultiSelect;
+        this.answerBitLengths = answerBitLengths;
+        this.answerLen = isMultiSelect.length;
+    }
+
+    public int generateCompatibilityScore() {
+        StringBuilder binaryScore = new StringBuilder();
+        StringBuilder binaryScoreSeparated = new StringBuilder();
+        for (int i = 0; i < this.answerLen; i++) {
+            String bundledBinary = answerSetToBinary(i);
+            binaryScore.append(bundledBinary);
+            if (this.isMultiSelect[i]) {
+                binaryScoreSeparated.append(bundledBinary).append("*");
+            } else{
+                binaryScoreSeparated.append(bundledBinary).append("-");
+            }
+        }
+        System.out.println("binaryScore: " + binaryScoreSeparated.toString());
+        return Integer.parseInt(binaryScore.toString(), 2);
```

Comment:
I understand that 2 is for parsing the integer in base 2 number system, but 2 can seem like a magic number? Might help to define it as a final variable like `BASE_TWO` and use `return Integer.parseInt(binaryScore.toString(), BASE_TWO);`

Reply:
I do agree with you, Helen. Documentation would be nice and it also ensures that we are understanding your code correctly, which allows us to test it correctly. Thus, including official test cases would be nice as well.

Reply:
Sounds good, fixing this in the next commit!